test/gurka/test_route.cc

```diff
@@ -821,3 +821,42 @@ TEST(MultipointRoute, WithPointsOnDeadends) {
     gurka::assert::raw::expect_path(result, {"AB", "BC", "BC"});
   }
 }
+
```

Comment:
Here's my shot at a test case.  It passes with and without my fix.  Without my fix the unidir A* leg enters the fallback logic and allows destination-only.  With my fix the unidir A* leg allows destination-only on the first pass.  I debugged everything I could think of (outside of doing something crazy like returning a new statistic) to author a case to prove my fix.

Reply:
IIUC, without your fix, the second leg (B->C) succeeds in the first pass (since it has `allow_destination_only=true` from the previous bidir astar run), but with your fix, the unidirectional astar succeeds only on the second try, correct? The number of passes is unfortunately not emitted out in the response, else we can use it to verify it in the test.

Can we make a map which has a different route if it uses a destination only edge somewhere in the middle? That way the 2 paths returned with  without the fix would be different?

Eg: in your current map, what if e introduce a destination-only edge somewhere on QP? Without your fix, it will use QP to reach C, but with your fix it should avoid it and take the edge further away (ZY).

Reply:
@mandeepsandhu You sort of have the gist of it.  Let me clarify a bit.

Old behavior:
Leg 1:  bidir A*.  Sets `allow_destination_only=false`.  Succeeds on first pass.
Leg 2:  unidir A*.  (Because leg 1 used bidir A*, `allow_destination_only` is still `false`.). Unidir A* fails to find a route on first pass.  Enters pass 2 logic, sets `allow_destination_only=true` and succeeds in second pass.

New behavior:
Leg 1:  bidir A*.  Sets `allow_destination_only=false`.  Succeeds on first pass.
Leg 2:  unidir A*.  Sets `allow_destination_only=true`.  Succeeds on first pass.

In both cases the result is successful and produce the same result.

I'm gonna ponder your suggestion a bit now...

Reply:
@mandeepsandhu Regarding your suggestion to have a destination edge along QP...

Having a destination edge in the middle of the route is supported but has a high transition cost so is avoided.

Even if I modify the cost of the intermediate destination edge transition cost to be 0.0, the same route is returned with/without my fix.  As mentioned above, without my fix it succeeds by entering the pass 2 logic, with my fix it succeeds in pass 1.

Good thoughts though, thanks.